libs/template/schema.go

```diff
@@ -0,0 +1,125 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/databricks/cli/libs/jsonschema"
+)
+
+// function to check whether a float value represents an integer
+func isIntegerValue(v float64) bool {
+	return v == float64(int(v))
+}
+
+// cast value to integer for config values that are floats but are supposed to be
+// integers according to the schema
+//
+// Needed because the default json unmarshaler for maps converts all numbers to floats
+func castFloatConfigValuesToInt(config map[string]any, jsonSchema *jsonschema.Schema) error {
+	for k, v := range config {
+		// error because all config keys should be defined in schema too
+		fieldInfo, ok := jsonSchema.Properties[k]
+		if !ok {
+			return fmt.Errorf("%s is not defined as an input parameter for the template", k)
+		}
+		// skip non integer fields
+		if fieldInfo.Type != jsonschema.IntegerType {
+			continue
+		}
+
+		// convert floating point type values to integer
+		valueType := reflect.TypeOf(v)
+		switch valueType.Kind() {
+		case reflect.Float32:
+			floatVal := v.(float32)
+			if !isIntegerValue(float64(floatVal)) {
+				return fmt.Errorf("expected %s to have integer value but it is %v", k, v)
+			}
+			config[k] = int(floatVal)
+		case reflect.Float64:
+			floatVal := v.(float64)
+			if !isIntegerValue(floatVal) {
+				return fmt.Errorf("expected %s to have integer value but it is %v", k, v)
+			}
+			config[k] = int(floatVal)
+		}
```

Comment:
Didn't catch this earlier; this should be a [type switch](https://go.dev/tour/methods/16).

```
	switch v := i.(type) {
	case int:
		fmt.Printf("Twice %v is %v\n", v, v*2)
	case string:
		fmt.Printf("%q is %v bytes long\n", v, len(v))
	default:
		fmt.Printf("I don't know about type %T!\n", v)
	}
```

Reply:
done